extensions/opc/src/putopc.cpp

```diff
@@ -0,0 +1,466 @@
+/**
+ * PutOPC class definition
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <memory>
+#include <string>
+#include <list>
+#include <map>
+#include <mutex>
+#include <thread>
+
+#include "opc.h"
+#include "putopc.h"
+#include "utils/ByteArrayCallback.h"
+#include "FlowFileRecord.h"
+#include "core/Processor.h"
+#include "core/ProcessSession.h"
+#include "core/Core.h"
+#include "core/Property.h"
+#include "core/Resource.h"
+#include "controllers/SSLContextService.h"
+#include "core/logging/LoggerConfiguration.h"
+#include "utils/Id.h"
+#include "utils/StringUtils.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+  core::Property PutOPCProcessor::ParentNodeID(
+      core::PropertyBuilder::createProperty("Parent node ID")
+          ->withDescription("Specifies the ID of the root node to traverse")
+          ->isRequired(true)->build());
+
+
+  core::Property PutOPCProcessor::ParentNodeIDType(
+      core::PropertyBuilder::createProperty("Parent node ID type")
+          ->withDescription("Specifies the type of the provided node ID")
+          ->isRequired(true)
+          ->withAllowableValues<std::string>({"Path", "Int", "String"})->build());
+
+  core::Property PutOPCProcessor::ParentNameSpaceIndex(
+      core::PropertyBuilder::createProperty("Parent node namespace index")
+          ->withDescription("The index of the namespace. Used only if node ID type is not path.")
+          ->withDefaultValue<int32_t>(0)->build());
+
+  core::Property PutOPCProcessor::ValueType(
+      core::PropertyBuilder::createProperty("Value type")
+          ->withDescription("Set the OPC value type of the created nodes")
+          ->isRequired(true)->build());
+
+  core::Property PutOPCProcessor::TargetNodeIDType(
```

Comment:
This should have the same validator as ParentNodeIDType.

Reply:
Can't as it's flowfile-dependent. :(

Reply:
😿